mod/groups/actions/groups/edit.php

```diff
@@ -210,21 +210,7 @@
 	));
 }
 
-$has_uploaded_icon = (!empty($_FILES['icon']['type']) && substr_count($_FILES['icon']['type'], 'image/'));
-
-if ($has_uploaded_icon) {
-	$filehandler = new ElggFile();
-	$filehandler->owner_guid = $group->owner_guid;
```

Comment:
I would have left a comment if it were so. The icon file hook takes care of this logic. I think this was some interim step in the process.